src/main/java/seedu/address/model/patient/Phone.java

```diff
@@ -30,7 +31,7 @@ public Phone(String phone) {
      * Returns true if a given string is a valid phone number.
      */
     public static boolean isValidPhone(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.matches(VALIDATION_REGEX) && test.length() <= 20;
```

Comment:
maybe you want to use a `private static final int MAX_PHONE_NUMBER_LENGTH = 20` instead of using magic constant?